lib/client-api/src/story_store.ts

```diff
@@ -406,6 +406,7 @@ export default class StoryStore {
         args: _stories[id].args,
         argTypes,
         globals: this._globals,
+        viewMode: this._selection?.viewMode,
```

Comment:
I might be overlooking possible scenarios here.
@ndelangen could you please give a small hand here? I'm not sure if it's ok to just get the value from `this._selection` here. It doesn't seem to be available at all times when this runs. Should it be?